clang/lib/Sema/SemaDecl.cpp

```diff
@@ -16412,28 +16438,53 @@ Decl *Sema::ActOnFinishFunctionBody(Decl *dcl, Stmt *Body, bool IsInstantiation,
         FD->getAttr<SYCLKernelEntryPointAttr>();
     if (FD->isDefaulted()) {
       Diag(SKEPAttr->getLocation(), diag::err_sycl_entry_point_invalid)
-          << SKEPAttr << /*defaulted function*/ 3;
+          << SKEPAttr << /*defaulted function*/ 2;
```

Comment:
I think we're at the point where this diagnostic should change to use `%enum_select` instead of `%select` so that we can use the enumerators here. WDYT?